### PR TITLE
config: Use write_mode from config

### PR DESCRIPTION
**Pull request by kamalmarhubi**
This commit tidies up handling of `write_mode` by setting it in the
config at the start, and removing the `write_mode` parameter threaded
throughout the formatting process.

### CODE REVIEW

File: src/bin/rustfmt.rs [hunk 1]
```diff
@@ -112,6 +113,14 @@ fn match_cli_path_or_file(config_path: Option<PathBuf>,
 fn update_config(config: &mut Config, matches: &Matches) {
     config.verbose = matches.opt_present("verbose");
     config.skip_children = matches.opt_present("skip-children");
+
+    let write_mode = matches.opt_str("write-mode").map(|ref s| {
+        WriteMode::from_str(s).expect(&format!("Invalid write-mode: {}", s))
```
Comment (nrc) [inline src/bin/rustfmt.rs]: We shouldn't panic due to invalid user input. We should exit gracefully with the message.

**Comment by nrc** [comment]
LGTM, just that one comment to address

File: src/bin/rustfmt.rs (hunk 1 above)
Comment (kamalmarhubi) [inline src/bin/rustfmt.rs]: Done.

**Comment by nrc** [comment]
I prefer the version you've currently implemented, thanks! However, you now have a test error.

Could you also squash the commits before I merge?

**Comment by kamalmarhubi** [comment]
> However, you now have a test error.

The formating tool that tests its own formatting... keeps biting! :-)

> Could you also squash the commits before I merge?

Should be all done!

**Comment by nrc** [comment]
Awesome, thanks!

**Pull request merged**